src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        if (processingNode == null) {
+            table[index] = new Node<K, V>(key, value, null);
+            size++;
+        } else {
+            while (processingNode.next != null) {
+                if (areKeysEqual(processingNode.key, key)) {
+                    processingNode.value = value;
+                    return;
+                }
+                processingNode = processingNode.next;
+            }
+            if (areKeysEqual(processingNode.key, key)) {
+                processingNode.value = value;
+                return;
+            }
+            processingNode.next = new Node(key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        while (processingNode != null) {
+            if (areKeysEqual(processingNode.key, key)) {
+                return processingNode.getValue();
+            } else {
+                processingNode = processingNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
+        return size;
+    }
+
+    private boolean areKeysEqual(K key, K current) {
+        return (current != null && current.equals(key))
```

Comment:
Previous comments about using Objects.equals are not resolved is there any issue?